test/fuzzing/mst_fuzz.cpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <memory>
+
+#include <gtest/gtest.h>
+#include <libfuzzer/libfuzzer_macro.h>
+
+#include "backend/protobuf/proto_transport_factory.hpp"
+#include "interfaces/iroha_internal/transaction_batch_factory_impl.hpp"
+#include "interfaces/iroha_internal/transaction_batch_parser_impl.hpp"
+#include "module/irohad/multi_sig_transactions/mst_test_helpers.hpp"
+#include "multi_sig_transactions/transport/mst_transport_grpc.hpp"
+
+using namespace testing;
+using namespace iroha::network;
+
+namespace fuzzing {
+  struct MstFixture {
+    std::shared_ptr<MstTransportGrpc> mst_transport_grpc_;
+
+    MstFixture() {
+      auto async_call_ = std::make_shared<
+          iroha::network::AsyncGrpcClient<google::protobuf::Empty>>();
+      std::unique_ptr<shared_model::validation::AbstractValidator<
+          shared_model::interface::Transaction>>
+          tx_validator =
+              std::make_unique<shared_model::validation::
+                                   DefaultOptionalSignedTransactionValidator>();
```

Comment:
The problem with this approach is that the possible changes in `application.cpp` will not be reflected in this file. Please think of a solution where components in `application.cpp` are referenced here.

Reply:
After discussion with @igor-egorov we decided to leave TODO and fix this issue later since it affect other tasks too.